scripts/generate-federate-match.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+set -eou pipefail
+shopt -s inherit_errexit
+
+! [ -x "$(command -v jq)" ] && echo 'jq not installed, the script requires it.' && exit 1
+! [ -x "$(command -v mimirtool)" ] && echo 'mimirtool not installed, the script requires it.' && exit 1
+! [ -x "$(command -v realpath)" ] && echo 'realpath not installed, the script requires it.' && exit 1
+! [ -x "$(command -v sort)" ] && echo 'sort not installed, the script requires it.' && exit 1
+! [ -x "$(command -v uniq)" ] && echo 'uniq not installed, the script requires it.' && exit 1
+! [ -x "$(command -v yq)" ] && echo 'yq not installed, the script requires it.' && exit 1
+
+function log() {
+    echo "${1:-}" >&2
+}
+
+function log_exit() {
+    log "${1:-}"
+
+    exit 1
+}
+
+function get_rules_metrics() {
+    local os_prom_rule_file="${1:-}"
+    [[ "${os_prom_rule_file}" = "" ]] && log_exit "Variable 'os_prom_rule_file' is empty."
+
+    local metrics_list_file="${2:-}"
+    [[ "${metrics_list_file}" = "" ]] && log_exit "Variable 'metrics_list_file' is empty."
+
+    local rules_file
+    rules_file=$(mktemp)
+
+    local json_file
+    json_file=$(mktemp)
+
+    yq '.spec' "${os_prom_rule_file}" > "${rules_file}"
+    mimirtool analyze rule-file "${rules_file}" --output="${json_file}"
+    jq '.ruleGroups[].metrics | select( . != null ) | .[]' "${json_file}" --raw-output >> "${metrics_list_file}"
+
+    rm -rf "${rules_file}"
+    rm -rf "${json_file}"
+}
+
+function append_consolidated_dashboard_fields() {
+    local federation_config_file="${1:-}"
+    [[ "${federation_config_file}" = "" ]] && log_exit "Variable 'federation_config_file' is empty."
+
+    yq --inplace '."match[]" += [ "container_memory_usage_bytes" ]' "${federation_config_file}"
```

Comment:
What is the connection between `container_memory_usage_bytes` and "consolidated dashboard"? How can we make sure this stays up-to-date going forward?

Reply:
I have removed this part.

I talked with Alex and Stephan - the consolidated dashboard is never finished. It is not used or maintained. I'll probably go forward and see to remove it.